src/main/java/net/pms/newgui/NavigationShareTab.java

```diff
@@ -816,80 +799,133 @@ public void actionPerformed(ActionEvent e) {
 				if (returnVal == JFileChooser.APPROVE_OPTION) {
-					((SharedFoldersTableModel) FList.getModel()).addRow(new Object[]{chooser.getSelectedFile().getAbsolutePath(), true});
-					if (FList.getModel().getValueAt(0, 0).equals(ALL_DRIVES)) {
-						((SharedFoldersTableModel) FList.getModel()).removeRow(0);
+					int firstSelectedRow = sharedFolders.getSelectedRow();
+					if (firstSelectedRow >= 0) {
+						((SharedFoldersTableModel) sharedFolders.getModel()).insertRow(
+							firstSelectedRow,
+							new Object[]{chooser.getSelectedFile().getAbsolutePath(), true}
+						);
+					} else {
+						((SharedFoldersTableModel) sharedFolders.getModel()).addRow(
+							new Object[]{chooser.getSelectedFile().getAbsolutePath(), true}
+						);
 					}
-					updateModel();
 				}
 			}
 		});
-		builderFolder.add(but, FormLayoutUtil.flip(cc.xy(1, 3), colSpec, orientation));
+		addButton.addMouseListener(new MouseAdapter() {
+			@Override
+			public void mouseClicked(MouseEvent e) {
+				if (configuration.isDefaultSharedFolders() && e.getButton() == MouseEvent.BUTTON1 && e.getClickCount() > 1) {
+					if (
+						JOptionPane.showConfirmDialog(
+							tmpsharedPanel,
+							Messages.getString("SharedFolders.EnableModify"),
+							null,
+							JOptionPane.YES_NO_OPTION,
+							JOptionPane.QUESTION_MESSAGE
+						) == JOptionPane.YES_OPTION
+					) {
+						toggleCustomizeFolders(tmpsharedPanel);
+					}
+				}
+			}
+		});
+		builderFolder.add(addButton, FormLayoutUtil.flip(cc.xy(2, 3), colSpec, orientation));
 
-		JImageButton but2 = new JImageButton("button-remove-folder.png");
-		but2.setToolTipText(Messages.getString("FoldTab.36"));
-		but2.addActionListener(new ActionListener() {
+		removeButton.setToolTipText(Messages.getString("FoldTab.36"));
+		removeButton.setEnabled(!defaultSharedFolders);
+		removeButton.addActionListener(new ActionListener() {
 			@Override
 			public void actionPerformed(ActionEvent e) {
-				if (FList.getSelectedRow() > -1) {
-					if (FList.getModel().getRowCount() == 0) {
-						folderTableModel.addRow(new Object[]{ALL_DRIVES, false});
-					} else {
-						PMS.get().getDatabase().removeMediaEntriesInFolder((String) FList.getValueAt(FList.getSelectedRow(), 0));
+				int[] rows = sharedFolders.getSelectedRows();
+				if (rows.length > 0) {
+					if (rows.length > 1) {
+						if (
+							JOptionPane.showConfirmDialog(
+								tmpsharedPanel,
+								String.format(Messages.getString("SharedFolders.ConfirmRemove"), rows.length),
+								Messages.getString("Dialog.Confirm"),
+								JOptionPane.YES_NO_OPTION,
+								JOptionPane.WARNING_MESSAGE
+							) != JOptionPane.YES_OPTION
+						) {
+							return;
+						}
+					}
+					for (int i = rows.length - 1; i >= 0; i--) {
+						((SharedFoldersTableModel) sharedFolders.getModel()).removeRow(rows[i]);
+						PMS.get().getDatabase().removeMediaEntriesInFolder((String) sharedFolders.getValueAt(sharedFolders.getSelectedRow(), 0));
```

Comment:
@SubJunk when the last item in the `sharedFolders` is deleted than here is thrown an error 

java.lang.ArrayIndexOutOfBoundsException: -1
	at java.util.Vector.elementData(Vector.java:737)
	at java.util.Vector.elementAt(Vector.java:480)
	at javax.swing.table.DefaultTableModel.getValueAt(DefaultTableModel.java:648)
	at javax.swing.JTable.getValueAt(JTable.java:2720)
	at net.pms.newgui.NavigationShareTab$38.actionPerformed(NavigationShareTab.java:858)
......